src/main/java/com/wise23/chariteed/controller/PatientGeneratorController.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.hl7.fhir.r4.model.Patient;
+
+import com.google.gson.*;
+import com.google.zxing.WriterException;
+
+import java.util.Random;
+import java.io.IOException;
+import java.util.List;
+
+@Controller
+public class PatientGeneratorController {
+
+    PatientService patientService = new PatientService();
+
+    @Autowired
+    UserService userService;
+
+    @GetMapping("admin/dashboard/generatePatient")
+    public String generatePatientForm(Model model) {
+        model.addAttribute("generatePatient", new PatientGenerator());
+        return "admin/dashboard/generatePatient";
+    }
+
+    @PostMapping("admin/dashboard/generatePatient")
+    public String generatePatientSubmit(@ModelAttribute PatientGenerator generator, Model model)
+            throws WriterException, IOException {
+        String id = Long.toString(generator.getId());
+        model.addAttribute("generatePatient", generator);
+
+        Patient patient = null;
+
+        // In case that the ID is not existing.
+        try {
+            patient = patientService.client.read().resource(Patient.class).withId(id).execute();
+        } catch (Exception e) {
+            System.out.println("ERROR: Patient ID does not exist.");
+            return "admin/dashboard/error";
+        }
+
+        String patientData = patientService.fhirContext.newJsonParser().setPrettyPrint(true)
+                .encodeResourceToString(patient);
+
+        generator.setPassword(generatePassword(patientData));
+
+        // QR code generation
+        QRCodeGenerator.createQRImage(id);
+
+        JsonObject name = JsonParser.parseString(patientData).getAsJsonObject().get("name").getAsJsonArray().get(0)
+                .getAsJsonObject();
+
+        User user = new User();
+        // Lot of dummy data
+        user.setFirstName(name.get("given").getAsJsonArray().get(0).getAsString());
+        user.setLastName(name.get("family").getAsString());
+        user.setMobile("2222222222");
+        user.setEmail("test@mail.de");
+        user.setPassword(generator.getPassword());
+        user.setRole(Role.USER);
+        user.setID(id);
+        user.setFile(generator.getFile());
+        List<Object> userPresentObj = userService.isUserPresent(user);
+        if ((Boolean) userPresentObj.get(0)) {
+            // TODO: Errorhandling
+            System.out.println("ERROR: Patient Account already exists!");
+            return "admin/dashboard/generatePatient";
+        }
+
+        userService.saveUser(user);
+        System.out.println("User " + user.getFirstName() + " created\n Password is: " + generator.getPassword());
+
+        return "admin/dashboard/patientGenerated";
+    }
+
+    private String generatePassword(String patientData) {
+        JsonObject name = JsonParser.parseString(patientData).getAsJsonObject().get("name").getAsJsonArray().get(0)
```

Comment:
Nah, I would first convert json object to a user object (or a userDTO - data transfer object) and then just use its fields

Reply:
How? The Json string looks something like this: 

`{
  "resourceType": "Patient",
  "id": "7175172",
  "meta": {
    "versionId": "1",
    "lastUpdated": "2022-12-02T12:25:25.475+00:00",
    "source": "#CL4cApvmtXCznO7f"
  },
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><table class=\"hapiPropertyTable\"><tbody/></table></div>"
  },
  "name": [ {
    "family": "Jameson",
    "given": [ "Jonah" ]
  } ],
  "gender": "male",
  "birthDate": "2022-01-01"
}
`
When I try to convert it to a user object like this:

`Gson gson = new Gson();
User user = gson.fromJson(patientData, User.class);`

It fails, probably because the fields differ.

Reply:
@JannesVolkens 

JSONObject jsonPatient = new JSONObject(patientData);
String patientName = jsonPatient.getString("name");

At least so.

Reply:
Done